### PR TITLE
Remove CTRE ifdef code for integration tests

### PR DESCRIPTION
**Pull request by Piphi5**
Also fixes a bug caught by the integration tests

### CODE REVIEW

**Comment by Piphi5** [comment]
Waiting on #255 to be merged

**Comment by calcmogul** [comment]
The commit message should also mention there's changes to some function arguments to fix lifetime issues. Ideally, that would be in a separate commit.

**Comment by calcmogul** [comment]
You should do an interactive rebase with each commit set to "edit" so you can check each commit compiles. That should help you track down the build issue.

File: sysid-projects/mechanism/src/main/cpp/Robot.cpp [hunk 1]
```diff
@@ -50,24 +50,19 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
...
+                         numSamples, controllerNames[0],
+                         m_controllers.front().get(), encoderInverted,
+                         encoderPorts, m_cancoder, m_revEncoderPort,
+                         m_revDataPort, m_encoder, m_position, m_rate);
   } catch (std::exception& e) {
     fmt::print("Project failed: {}\n", e.what());
     std::exit(-1);
   }
   std::cout.flush();
-#ifdef INTEGRATION
```
Comment (calcmogul) [inline sysid-projects/mechanism/src/main/cpp/Robot.cpp:66]: This #ifdef shouldn't be removed.

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**